include/zephyr/kernel.h

```diff
@@ -3416,6 +3416,81 @@ static inline unsigned int z_impl_k_sem_count_get(struct k_sem *sem)
 
 /** @} */
 
+#ifdef CONFIG_SCHED_IPI_SUPPORTED
+struct k_ipi_work;
+typedef void (*k_ipi_func_t)(struct k_ipi_work *work);
+
+struct k_ipi_work {
+	sys_dnode_t        node[CONFIG_MP_MAX_NUM_CPUS] ;   /* Node in IPI work queue */
+	k_ipi_func_t   func;     /* Function to execute on target CPU */
+	struct k_event event;    /* Event to signal when processed */
+	uint32_t       bitmask;  /* Bitmask of targeted CPUs */
+};
```

Comment:
please add doxygen for the struct and typedef as well -- if the fields of the struct are meant to be API they need Doxygen comments too (not just simple C-style comments), and if they are meant to be internal then wrap in a `@cond INTERNAL_HIDDEN / @endcond` block.